zerver/tests/test_signup.py

```diff
@@ -736,20 +738,21 @@ def test_login_nonexist_user(self) -> None:
         self.assert_logged_in_user_id(None)
 
     def test_login_wrong_subdomain(self) -> None:
+        email = self.mit_email("sipbtest")
         with self.assertLogs(level="WARNING") as m:
-            result = self.login_with_return(self.mit_email("sipbtest"), "xxx")
+            result = self.login_with_return(email, "xxx")
             self.assertEqual(
                 m.output,
                 [
                     "WARNING:root:User sipbtest@mit.edu attempted password login to wrong subdomain zulip"
                 ],
             )
         self.assertEqual(result.status_code, 200)
-        self.assert_in_response(
-            "Your Zulip account is not a member of the "
-            "organization associated with this subdomain.",
-            result,
+        expected_error = (
+            f"Your Zulip account {email} is not a member of the "
+            + "organization associated with this subdomain."
```

Comment:
I kinda feel like this "wrong subdomain" error category should be removed; it feels like we should just tell the user it was a failed login rather than leaking their credentials were correct but for the wrong domain.  (At least on the web; I could see e.g. emailing the user).